contentcuration/contentcuration/assets/channelList/views/ChannelSet/ChannelSetModal.vue

```diff
@@ -0,0 +1,354 @@
+          >
+            {{ $tr('saveButtonLabel') }}
+          </VBtn>
+          <span
+            v-show="saving"
+            class="spinner pull-right"
+          >
+          </span>
+          <span v-show="error" class="red-text error-text pull-right">
+            {{ $tr('errorText') }}
+          </span>
+        </div>
+      </VCardActions>
+    </VCard>
+  </VDialog>
+</template>
+
+
+<script>
+
+  import { mapGetters, mapActions, mapMutations, mapState } from 'vuex';
+  import sortBy from 'lodash/sortBy';
+  import CopyToken from 'edit_channel/sharedComponents/CopyToken.vue';
+  import { isDummyId } from '../../utils';
+  import { RouterNames } from '../../constants';
+
+  export default {
+    name: 'ChannelSetModal',
+    components: {
+      CopyToken,
+    },
+    props: {
+      channelSetId: {
+        type: String,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        saving: false,
+        loadingChannels: true,
+        expansionPanel: 0,
+      };
+    },
+    $trs: {
+      newSetHeader: 'New Collection',
+      editingSetHeader: 'Editing Collection',
+      cancelButtonLabel: 'Close',
+      saveButtonLabel: 'Save',
+      saveCloseButtonLabel: 'Save & Close',
+      noChangesTitle: 'No changes detected',
+      invalidCollection: 'Must enter all required fields',
+      saving: 'Saving...',
+      errorText: 'There was a problem saving your collection',
+      loading: 'Loading...',
+      titleLabel: 'Title',
+      titlePlaceholder: 'Title your collection',
+      descriptionLabel: 'Description',
+      descriptionPlaceholder: 'Describe your collection',
+      selectButtonLabel: 'Select',
+      backButtonLabel: 'Back to details',
+      channelCountText:
+        '{channelCount, plural, =1 {# channel in your collection} other {# channels in your collection}}',
+      titleRequiredText: 'Title is required',
+      charCount: '{charCount, plural, =1 {# character left} other {# characters left}}',
+      versionText: 'Version {version}',
+      unpublishedTitle: '{channelName} must be published to import it into Kolibri',
+      publishedChannels: 'Published channels that can be imported into Kolibri',
+      unpublishedChannels: 'These channels must be published to import them into Kolibri',
+    },
+    created() {
+      this.loadChannels();
+    },
+    computed: {
+      ...mapGetters('channelSet', [
+        'getChannelSet',
+        'getChannelSetIsDirty',
+        'getChannelSetIsValid',
+      ]),
+      ...mapGetters('channelList', { availableChannels: 'channels' }),
+      ...mapState('channelSet', ['error']),
+      modalTitle() {
+        if (this.isNewSet) {
+          return this.$tr('newSetHeader');
+        }
+        return this.$tr('editingSetHeader');
+      },
+      open: {
+        get() {
+          return true;
+        },
+        set(open) {
+          if (!open) {
+            this.close();
+          };
+        },
+      },
+      name: {
```

Comment:
> Here we use a pattern of computed prop getters and setters to make them compatible with `v-model`

This works, but the computed prop getters/setters always seemed a little strange looking to me. This might be done more conventionally by binding to simple computed props and methods:

```html
<VTextField
  :value="name"
  @input="updateName"
/>
```

```js
computed: {
  name() {
    return this.channelSet.name || '';
  },
},
methods: {
  updateName(value) {
    this.updateChannelSet({ id: this.channelSetId, name: value });
  },
},
```

See also https://github.com/learningequality/kolibri/issues/4499

Reply:
Getters and setters seem more natural to me, because that's the ES5 way, whether we use `value.sync` or v-model on top of that is less of an issue for me, but I think that `VTextField` would still need to use v-model because of the way it is constructed (although I've not looked at the implementation, so maybe it is `.sync` compatible too).